plugin/src/main/java/org/opensearch/ml/task/MLPredictTaskRunner.java

```diff
@@ -336,12 +347,60 @@ private void runPredict(
                     if (mlInput.getAlgorithm() == FunctionName.REMOTE) {
                         long startTime = System.nanoTime();
                         ActionListener<MLTaskResponse> trackPredictDurationListener = ActionListener.wrap(output -> {
+
                             if (output.getOutput() instanceof ModelTensorOutput) {
                                 validateOutputSchema(modelId, (ModelTensorOutput) output.getOutput());
                             }
-                            handleAsyncMLTaskComplete(mlTask);
-                            mlModelManager.trackPredictDuration(modelId, startTime);
-                            internalListener.onResponse(output);
+                            if (mlTask.getTaskType().equals(MLTaskType.BATCH_PREDICTION)) {
+                                Map<String, Object> transformJob = new HashMap<>();
+                                ModelTensorOutput tensorOutput = (ModelTensorOutput) output.getOutput();
+                                if (tensorOutput != null
+                                    && tensorOutput.getMlModelOutputs() != null
+                                    && !tensorOutput.getMlModelOutputs().isEmpty()) {
+                                    ModelTensors modelOutput = tensorOutput.getMlModelOutputs().get(0);
+                                    if (modelOutput.getMlModelTensors() != null && !modelOutput.getMlModelTensors().isEmpty()) {
+                                        Map<String, Object> dataAsMap = (Map<String, Object>) modelOutput
+                                            .getMlModelTensors()
+                                            .get(0)
+                                            .getDataAsMap();
+                                        if (dataAsMap != null
+                                            && (dataAsMap.containsKey("TransformJobArn") || dataAsMap.containsKey("id"))) {
+                                            transformJob.putAll(dataAsMap);
```

Comment:
I think you can just convert all the Object to String for easier data handling in the Get Task action, without any loss.

Reply:
Okay. Let me try to make that change and verify